### PR TITLE
Checking propetry for null changed from Or to OrElse

### PR DESCRIPTION
**Pull request by Ilchert**
Checking propetry for null chanaged from "($it.Property == null) | false" (bitwise OR) to "($it.Property == null) || false" (logical OR).

### CODE REVIEW

File: OData/src/System.Web.OData/OData/Query/Expressions/FilterBinder.cs [hunk 1]
```diff
@@ -1779,7 +1779,7 @@ private static Expression CheckIfArgumentsAreNull(Expression[] arguments)
             if (arguments.Any())
```
Comment (VikingsFan) [inline OData/src/System.Web.OData/OData/Query/Expressions/FilterBinder.cs]: Please add/modify unit test for FilterBinder's change in FilterBinderTest, make sure all unit test is pass.
FYI http://odata.github.io/WebApi/#09-01-unittest-e2etest

**Comment by Ilchert** [comment]
I added Unit test.

**Comment by VikingsFan** [comment]
Thanks for your contribution! Merged 35b7f5c3c7271ee2cc0b8c4eedafd5f197d91ba5 & 2bcbe4c81633f944b5866217467a8eb8deb9b50c

Please refine unit test next time, I refine for you in https://github.com/OData/WebApi/commit/90e39a6d2c62bdb3b5fa528c25dfb40e2de9cce5#diff-7e0105e342476f0d5cdac59a7217476b

**Comment by congysu** [comment]
Thanks @Ilchert for your contribution.  Can you help sign the CLA via the email you received, or via https://cla.microsoft.com.

**Comment by congysu** [comment]
Pending CLA.

**Comment by Ilchert** [comment]
Sorry for delay with CLA.

**Comment by congysu** [comment]
Thanks @Ilchert!

**Pull request closed without merging**